src/main/java/net/minecraftforge/items/itemhandlerobserver/IItemHandlerObservable.java

```diff
@@ -0,0 +1,19 @@
+package net.minecraftforge.items.itemhandlerobserver;
+
+
+public interface IItemHandlerObservable
+{
+    /**
+     * @param observer adds a observer to this
+     * @return true when the observer is added
+     */
+    boolean addObserver(IItemHandlerObserver observer);
+
+    void removeObserver(IItemHandlerObserver observer);
+
+    boolean containsObserver(IItemHandlerObserver observer);
```

Comment:
This might be a bit superfluous. An external user should not have to deal with registering duplicated listeners or removing non nonexistent ones. Just have add/remove ignore these cases. And document that duplicate observer instances are not allowed.